src/external_cli/arg_builder.rs

```diff
@@ -0,0 +1,208 @@
+/// A helper to make passing arguments to [`Command`](std::process::Command) more convenient.
+#[derive(Debug, Clone)]
+pub struct ArgBuilder(Vec<String>);
+
+impl ArgBuilder {
+    /// Create a new builder for command arguments.
+    pub const fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    /// Add a singular argument.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// ArgBuilder::new().arg("--release");
+    /// ```
+    pub fn arg<A>(mut self, arg: A) -> Self
+    where
+        A: Into<String>,
+    {
+        self.0.push(arg.into());
+        self
+    }
+
+    /// Add an argument with a value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// ArgBuilder::new().add_with_value("--bin", "bevy");
+    /// ```
+    pub fn add_with_value<A, V>(self, arg: A, value: V) -> Self
+    where
+        A: Into<String>,
+        V: Into<String>,
+    {
+        self.arg(arg).arg(value)
+    }
+
+    /// Add a boolean flag with the given name, if `value` is `true`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// let is_release = true;
+    /// ArgBuilder::new().add_flag_if("--release", is_release);
+    /// ```
+    pub fn add_flag_if<N>(self, name: N, value: bool) -> Self
+    where
+        N: Into<String>,
+    {
+        if value {
+            self.arg(name)
+        } else {
+            self
+        }
+    }
+
+    /// Add an argument with an optional value.
```

Comment:
```suggestion
    /// Add an argument with an optional value.
    ///
    /// If value is `&None`, no argument will be added.
```

I originally interpreted this documentation differently. I thought:

```rust
add_opt_value("foo", &None) == "--foo"
```

But it actually is:

```rust
add_opt_value("foo", &None) == ""
```

I hope these documentation additions will clarify this a bit more. :)